lib/vsc/mympirun/mpi/mpi.py

```diff
@@ -831,26 +830,38 @@ def make_mpiexec_hydra_options(self):
             else:
                 self.log.debug("make_mpiexec_hydra_options: no rmk from HYDRA_RMK %s and hydra_info %s",
                                self.HYDRA_RMK, self.hydra_info)
+
+        default_launcher = getattr(self, 'HYDRA_LAUNCHER', None)
+        if self.options.launcher:
+            launcher = [self.options.launcher]
+            launcher.extend(default_launcher)
         else:
-            launcher = None
-            if getattr(self, 'HYDRA_LAUNCHER', None) is not None:
-                launcher = [x for x in self.HYDRA_LAUNCHER if x in self.hydra_info.get('launcher', [])]
-                if launcher:
-                    self.log.debug("make_mpiexec_hydra_options: HYDRA: launcher %s, using first one", launcher)
-                else:
-                    self.log.debug("make_mpiexec_hydra_options: no launcher from HYDRA_LAUNCHER %s and hydra_info %s",
-                                   self.HYDRA_LAUNCHER, self.hydra_info)
+            launcher = default_launcher
+
+        if launcher:
```

Comment:
at least print a debug statement in an `else` here that says something like `No custom launcher specified`?